### PR TITLE
Apply the cleansing fire

### LINKED ISSUE #950: Rip out the shared process
🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 

The shared process was a bit of over-engineering on my part. The idea was that it was the central place where multiple windows would coordinate our persisted data. But:

1. We're not supporting multiple windows for 1.0.
2. We've been really inconsistent in using it, so that if/when we _do_ want to support multiple windows, we'll need to rethink this approach anyways.

### PR DESCRIPTION
**Pull request by joshaber**
giphy-2

Fixes #950

### CODE REVIEW

**Comment by gyg3** [comment]
Op

**Comment by joshaber** [comment]
This is ready for review ⚡️ 🔥 👨‍🚒

File: app/src/lib/dispatcher/accounts-store.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        account.name
+      )
+      const token = await this.secureStore.getItem(
+        getKeyForAccount(accountWithoutToken),
+        account.login
+      )
+      return accountWithoutToken.withToken(token || '')
+    })
+
+    this.accounts = await Promise.all(accountsWithTokens)
```
Comment (joshaber) [inline app/src/lib/dispatcher/accounts-store.ts]: Good shout 👍

File: app/src/lib/dispatcher/app-store.ts [hunk 2]
```diff
@@ -2204,4 +2252,125 @@ export class AppStore {
...
+      const validatedPath = await validatedRepositoryPath(path)
+      if (validatedPath) {
+        const addedRepo = await this.repositoriesStore.addRepository(
+          validatedPath
+        )
+        const refreshedRepo = await this.refreshGitHubRepositoryInfo(addedRepo)
+        addedRepositories.push(refreshedRepo)
+      } else {
+        const error = new Error('add-repository')
+        error.message = `${path} isn't a git repository.`
```
Comment (joshaber) [inline app/src/lib/dispatcher/app-store.ts]: lol I have no idea

**Comment by joshaber** [comment]
giphy-2

**Comment by joshaber** [comment]
source

**Pull request merged**